pkg/disruption/backend/sampler/remote.go

```diff
@@ -0,0 +1,501 @@
+	if err != nil {
+		return err
+	}
+	apiIntHost := strings.Replace(internalAPI.Hostname(), "api.", "api-int.", 1)
+
+	projectClient, err := projectclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	err = callProject(ctx, projectClient, true)
+	if err != nil {
+		return err
+	}
+	err = callServiceAccount(ctx, kubeClient, true)
+	if err != nil {
+		return err
+	}
+	err = callRBACClusterAdmin(ctx, kubeClient, true)
+	if err != nil {
+		return err
+	}
+	err = callRBACHostaccess(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	return callMonitorDaemonset(ctx, kubeClient, apiIntHost, true)
+}
+
+func deleteTestBed(ctx context.Context, projectClient *projectclient.Clientset, kubeClient *kubernetes.Clientset) error {
+	// Stop daemonsets first so that test stop before serviceaccount is removed
+	// and permission issues from apiserver are not recorded as disruption
+	err := callMonitorDaemonset(ctx, kubeClient, "", false)
+	if err != nil {
+		return err
+	}
+
+	err = callRBACClusterAdmin(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	err = callRBACHostaccess(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	err = callServiceAccount(ctx, kubeClient, false)
+	if err != nil {
+		return err
+	}
+	return callProject(ctx, projectClient, false)
+}
+
+func createDaemonset(ctx context.Context, clientset *kubernetes.Clientset, create bool, obj *appsv1.DaemonSet) error {
+	client := clientset.AppsV1().DaemonSets(namespace)
+	var err error
+	if create {
+		_, err = client.Create(ctx, obj, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		if err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 5*time.Minute, true, func(context.Context) (bool, error) {
+			ds, err := client.Get(context.Background(), obj.Name, metav1.GetOptions{})
+			if err != nil {
+				framework.Logf("error getting daemonsets %v", err)
+				return false, nil
+			}
+			return ds.Status.NumberReady > 0, nil
+		}); err != nil {
+			return fmt.Errorf("daemonset %s didn't roll out: %v", obj.Name, err)
+		}
+
+	} else {
+		err = client.Delete(ctx, obj.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+		if err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 5*time.Minute, true, func(context.Context) (bool, error) {
+			_, err := client.Get(context.Background(), obj.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return true, nil
+				}
+				framework.Logf("Error getting daemonset %s: %v", err)
+				return false, nil
+			}
+			return false, nil
+		}); err != nil {
+			return fmt.Errorf("daemonset %s didn't roll out: %v", obj.Name, err)
+		}
+	}
+
+	return err
+}
+
+func callMonitorDaemonset(ctx context.Context, clientset *kubernetes.Clientset, apiIntHost string, create bool) error {
+	name := "pod-monitor"
+	labels := map[string]string{
+		"app": name,
+	}
+	bTrue := true
+	obj := &appsv1.DaemonSet{
```

Comment:
create these as manifests and use embed. That allows us to use them from the CLI and read them easily.